Create containsPredicate() API

smukil opened the pull request:
Checks the message bodies (i.e. the data in the hash map), and
returns true on the first match with 'predicate'.

rsrinivasanNetflix committed:
Adding simple demo for queues

rsrinivasanNetflix committed:
Trying to fix flaky test

rsrinivasanNetflix committed:
Fixes from comments

rsrinivasanNetflix committed:
Added doc comment

smukil committed:
Create containsPredicate() API

Checks the message bodies (i.e. the data in the hash map), and
returns true on the first match with 'predicate'.

smukil committed:
Create containsPredicate() API

Checks the message bodies (i.e. the data in the hash map), and
returns true on the first match with 'predicate'.

rsrinivasanNetflix committed:
Merge pull request #40 from Netflix/feature/addSimpleDemo

Adding simple demo for queues

rsrinivasanNetflix commented on dyno-queues-core/src/main/java/com/netflix/dyno/queues/DynoQueue.java:
```diff
@@ -115,7 +115,19 @@
+	 * Checks the message bodies (i.e. the data in the hash map), and returns true on the first match with
+	 * 'predicate'.
+	 *
+	 * Disclaimer: This is a potentially expensive call, since we will iterate over the entire hash map in the
```
Maybe also add that it will slow down other accesses to the cluster while this is running.

rsrinivasanNetflix commented on dyno-queues-redis/src/main/java/com/netflix/dyno/queues/redis/v2/RedisPipelineQueue.java:
```diff
@@ -467,6 +467,9 @@ public boolean ensure(Message message) {
     }
 
+    @Override
+    public boolean containsPredicate(String predicate) { throw new UnsupportedOperationException(); }
```
since the pipeline queue doesnt use pipelining for anything except push and pop, maybe we can add the same functionality here also?

smukil committed:
Cast from 'JedisCommands' to 'DynoJedisClient' instead of 'Jedis' in
containsPredicate()

The dyno-queues are always built using a 'DynoJedisClient' object and
not a 'Jedis' object. This patch fixes a bug in the previous patch that
would throw a runtime error if a DynoJedisClient object was used to
build the queue.

smukil committed:
Merge branch 'containsPredicate_api' of https://github.com/smukil/dyno-queues into containsPredicate_api

smukil commented:
Messed up the rebase. Will close this and open a new PR
